glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+                (475.0, 344.0),
+            );
+            path.line_to((588.0, 407.0));
+            path.line_to((567.0, 260.0));
+            path.curve_to((567.0, 414.0), (464.0, 510.0), (316.0, 510.0));
+            path.line_to((316.0, 437.0));
+            path.close_path();
+            path
+        }
+
+        pub(super) fn dotabove_ar() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((58.0, -58.0));
+            path.curve_to((90.0, -58.0), (116.0, -32.0), (116.0, 0.0));
+            path.curve_to((116.0, 32.0), (90.0, 58.0), (58.0, 58.0));
+            path.curve_to((26.0, 58.0), (0.0, 32.0), (0.0, 0.0));
+            path.curve_to((0.0, -32.0), (26.0, -58.0), (58.0, -58.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn sofiaSans() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((190.0, 327.0));
+            path.line_to((199.0, 497.0));
+            path.line_to((199.0, 488.0));
+            path.line_to((32.0, 503.0));
+            path.line_to((190.0, 327.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn largeCrossing() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((378.0, 615.0));
+            path.line_to((344.0, 706.0));
+            path.line_to((444.0, 660.0));
+            path.line_to((281.0, 699.0));
+            path.line_to((378.0, 615.0));
+            path.close_path();
+            path
+        }
+    }
+
+    macro_rules! assert_approx {
+        ($left:expr, $right:expr) => {
+            assert_approx!($left, $right, 1e-4)
+        };
+        ($left:expr, $right:expr, $accuracy:literal) => {
+            assert!(
+                ($left - $right).abs() < $accuracy,
+                "{} !~= {}",
+                $left,
+                $right
+            )
+        };
+    }
+
+    #[test]
+    fn test_empty_glyph() {
+        let glyph = python_test_glyphs::space();
+        let reveresed = glyph.reverse_subpaths();
+        for g in [glyph, reveresed] {
+            assert!(erase_open_corners(&g).is_none());
+        }
+    }
+
+    #[test]
+    fn test_corner_glyph() {
+        let glyph = python_test_glyphs::hasCornerGlyph();
+        let after = erase_open_corners(&glyph).unwrap();
+        let new_pt = after.segments().nth(1).unwrap().end();
+
+        assert_approx!(new_pt.x, 114.5417);
+        assert_approx!(new_pt.y, 191.2080);
+    }
+
+    #[test]
+    fn test_curve_curve_glyph() {
+        let glyph = python_test_glyphs::curvyCornerGlyph();
+        let after = erase_open_corners(&glyph);
+        let new_pt = after.unwrap().segments().next().unwrap().start();
+        // teaching to the test: because our segment splitting impl is different
+        // from fonttools', we get slightly different values here; we might get
+        // a few off-by-ones but that's okay?
```

Comment:
off by 0.2 seems like kind of a lot?

Reply:
I agree, but I didn't want to try and port the fonttools impl just for the sake of matching... ultimately any difference is going to leave open the possibility of rounding errors, and i think this is a case where if we very occasionally use up one of our off-by-one budget here it's not the end of the world?

Reply:
I still find the magnitude alarming. If you told me we are off by 0.001 ... sure, but 0.2 seems significant unless we modified the implementation in a way that we expect to produce a large increase in the error rate? Or conversely we believe fonttools accumulates more error than we do somehow?

Reply:
okay your instincts here were good, i'd missed something that the glyphsLib impl was doing (explicitly rebuilding the path after finishing all the operations) and so we were inadvertently ending up with a closed path where the start point was different from the end point of the last segment; fixing that means we can remove that accuracy parameter completely.